src/fields/ntt.rs

```diff
@@ -0,0 +1,268 @@
+use crate::fields::FieldElement;
+
+/// Represents an element of an NTT-friendly field.
+///
+/// Fields implementing this trait must have a subgroup under multiplication with
+pub trait NttFieldElement: FieldElement {
+    /// A 2^k-th root of unity in the field.
+    ///
+    /// A 2^k-th root of unity of a field is a generator of the subgroup of the multiplicative group
+    /// with order 2^k. Thus, it satisfies omega^(2^k) = 1, and omega^i != 1 for 0 < i < 2^k.
```

Comment:
What's `k` in this context? Also, it would help to explain that `ROOT_OF_UNITY = omega`.